### ISSUE TITLE
API is (mostly) inaccessible on Windows

### ISSUE THREAD

**Issue by Mithgol** [ipfs/go-ipfs#1758]
The command `ipfs swarm peers` produces the response `Error: Post http://127.0.0.1:5001/api/v0/swarm/peers?encoding=json&stream-channels=true: read tcp 127.0.0.1:55749->127.0.0.1:5001: wsarecv: An existing connection was forcibly closed by the remote host.`

The `ipfs daemon` is running (in another window) and IPFS is unblocked in Windows firewall.

**Comment by jbenet** [comment]
is this always, or some times?

**Comment by Mithgol** [comment]
Looks like always.

Of course `0.3.8-dev` [build](https://gobuilder.me/github.com/ipfs/go-ipfs/cmd/ipfs) is only ≈9 hours old and thus I might have missed some chance to encounter a successful `ipfs swarm peers`.

But I don't think so.

**Comment by Mithgol** [comment]
The next useful question would be: “Is it just me, or do other people on Windows experience the same issue?”

Quite fortunately, on GitHub it's possible to notify and summon people by mentioning them. I am mentioning @slothbag and @mjanczyk from #1661 and @vegberg from #1681; they'd no longer encounter neither #1661 nor #1681 in the new 0.3.8-dev, they'd be eager to try 0.3.8-dev, and thus they are likely to encounter the same problem that I've described above.

Chime in. Is it my problem here or some more common problem?

**Comment by mjanczyk** [comment]
It hasn't occurred for me with my own build (git master).
However, I downloaded gobuilder binary and started getting this error, for less than 50% of "ipfs swarm peers" executions. So I rebuilded ipfs by myself, using go 1.5.1 instead of 1.4.2, and it started occurring with that binary. Builded with 1.4.2 again - again could not reproduce, not even once.

Still, it occurs only on every other command execution. Doesn't seem to depend on a version of ipfs.exe I use to run daemon, only on client build. Does never occur just after daemon boot, with low peer count, but it doesn't look like I'm hitting some max value, neither - after connecting to more peers, every other call that is successful outputs more lines.

Idk if it could really be compiler version related?

**Comment by Mithgol** [comment]
I've just repeated `ipfs swarm peers` ten times in quick succession, and I've got:

* 8 times `Error: Post http://127.0.0.1:5001/api/v0/swarm/peers?encoding=json&stream-channels=true: read tcp 127.0.0.1:49848->127.0.0.1:5001: wsarecv: An existing connection was forcibly closed by the remote host.`

* once `Error: read tcp 127.0.0.1:49857->127.0.0.1:5001: wsarecv: An existing connection was forcibly closed by the remote host.`

* once the list of peers

So it's a probabilistic thing, contrary to what I've thought 11 hour ago.

Even on my computer it works sometimes.

**Comment by jbenet** [comment]
This may be the HTTP trailers problem. Maybe we could merge that fix dependent on the go compiler version (the issue is that fix _requires_ go 1.5, and I wanted to give some time to air out weirdness with 1.5, like this)



—

**Comment by Mithgol** [comment]
Which fix?

**Comment by mjanczyk** [comment]
@Mithgol I would presume #1621. Patch seems to be working, too, afaict and unless I mis-tested.

**Comment by whyrusleeping** [comment]
This should be resolved now, open a new issue if problems reoccur

**Comment by Mithgol** [comment]
Yes, resolved long ago.

I've just forgotten that the issue's been still open.
